test/e2e/framework/globalegressip.go

```diff
@@ -0,0 +1,85 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+var globalEgressIPGVR = &schema.GroupVersionResource{
+	Group:    "submariner.io",
+	Version:  "v1",
+	Resource: "globalegressips",
+}
+
+func globalEgressIPClient(cluster ClusterIndex, namespace string) dynamic.ResourceInterface {
+	return DynClients[cluster].Resource(*globalEgressIPGVR).Namespace(namespace)
+}
+
+func CreateGlobalEgressIP(cluster ClusterIndex, obj *unstructured.Unstructured) error {
+	geipClient := globalEgressIPClient(cluster, obj.GetNamespace())
+	AwaitUntil("create GlobalEgressIP", func() (interface{}, error) {
+		egressIP, err := geipClient.Create(context.TODO(), obj, metav1.CreateOptions{})
+		if apierrors.IsAlreadyExists(err) {
+			err = geipClient.Delete(context.TODO(), obj.GetName(), metav1.DeleteOptions{})
+			if err != nil {
+				fmt.Printf("Returning error while deleting: %v", err)
+				return nil, err
+			}
+
+			egressIP, err = geipClient.Create(context.TODO(), obj, metav1.CreateOptions{})
+		}
+
+		return egressIP, err
+	}, NoopCheckResult)
+
+	return nil
+}
+
+func AwaitGlobalEgressIPs(cluster ClusterIndex, name, namespace string) []string {
+	geipClient := globalEgressIPClient(cluster, namespace)
+	obj := AwaitUntil(fmt.Sprintf("await GlobalEgressIP %s", name),
+		func() (interface{}, error) {
+			resGip, err := geipClient.Get(context.TODO(), name, metav1.GetOptions{})
+			if apierrors.IsNotFound(err) {
+				return nil, nil
+			}
+			return resGip, err
+		},
+		func(result interface{}) (bool, string, error) {
+			if result == nil {
+				return false, fmt.Sprintf("GlobalEgressIP %s not found yet", name), nil
+			}
+
+			globalIPs := getGlobalIPs(result.(*unstructured.Unstructured))
+			if len(globalIPs) == 0 {
+				return false, fmt.Sprintf("GlobalEgressIP %q exists but allocatedIPs not available yet", name), nil
+			}
+			return true, "", nil
+		})
+
+	return getGlobalIPs(obj.(*unstructured.Unstructured))
```

Comment:
This is essentially identical to `AwaitClusterGlobalEgressIPs`.  We can create a generalized function: 
    
```
func AwaitAllocatedEgressIPs(client dynamic.ResourceInterface, name string) []string {
    obj := AwaitUntil(fmt.Sprintf("await allocated egress IPs for %q", name),
		func() (interface{}, error) {
			obj, err := client.Get(context.TODO(), name, metav1.GetOptions{})
			...
		},
		func(result interface{}) (bool, string, error) {
			if result == nil {
				return false, fmt.Sprintf("Egress IP resource %q not found yet", name), nil
			}

			globalIPs := getGlobalIPs(result.(*unstructured.Unstructured))
			if len(globalIPs) == 0 {
				return false, fmt.Sprintf("Egress IP resource %q exists but allocated IPs not available yet", name), nil
			}
			return true, "", nil
		})

	return getGlobalIPs(obj.(*unstructured.Unstructured))
}

```